diff --git a/Config/ip_filter.conf b/Config/ip_filter.conf
new file mode 100644
index 000000000..dfe20e028
--- /dev/null
+++ b/Config/ip_filter.conf
@@ -0,0 +1,2 @@
+172.27.112.1
+185.125.190.56
\ No newline at end of file
diff --git a/Config/test_captured_trafic.pcap b/Config/test_captured_trafic.pcap
new file mode 100644
index 000000000..ebf2ae295
Binary files /dev/null and b/Config/test_captured_trafic.pcap differ
diff --git a/src/decode-ipv4.c b/src/decode-ipv4.c
index 8db290c94..d35b48fba 100644
--- a/src/decode-ipv4.c
+++ b/src/decode-ipv4.c
@@ -38,6 +38,39 @@
 #include "flow.h"
 #include "util-print.h"
 
+#define MAX_FILTER_IPS 100
+#define MAX_IP_LEN 16
+
+static char allowed_ips[MAX_FILTER_IPS][MAX_IP_LEN];
+static int allowed_count = 0;
+
+
+/* Загрузка разрешенных IP из файла конфигурации */
+void LoadAllowedIPs(const char *path)
+{
+    FILE *f = fopen(path, "r");
+    if (!f) {
+        printf("Warning: couldn't open IP filter file: %s \n", path );
+        return;
+    }
+    char line[MAX_IP_LEN];
+    while (fgets(line, sizeof(line), f) && allowed_count < MAX_FILTER_IPS) {
+        line[strcspn(line, "\n")] = 0; // убрать \n
+        if (strlen(line) > 0)
+            strncpy(allowed_ips[allowed_count++], line, MAX_IP_LEN); // копируем в массив разрешённых IP
+    }
+    fclose(f);
+}
+/* Проверка, есть ли IP в списке разрешённых */
+bool IsAllowedIP(const char *ip)
+{
+    for (int i = 0; i < allowed_count; i++) {
+        if (strcmp(allowed_ips[i], ip) == 0) //сравниваем строки
+            return true;
+    }
+    return false;
+}
+
 /* Generic validation
  *
  * [--type--][--len---]
@@ -499,6 +532,35 @@ static const IPV4Hdr *DecodeIPV4Packet(Packet *p, const uint8_t *pkt, uint16_t l
     SET_IPV4_SRC_ADDR(ip4h, &p->src);
     SET_IPV4_DST_ADDR(ip4h, &p->dst);
 
+/* CUSTOM: log src/dist */
+
+
+
+char src[16], dst[16];
+
+PrintInet(AF_INET, (const void *)GET_IPV4_SRC_ADDR_PTR(p), src, sizeof(src)); // использую PrintInet() для преобразования IP-адресов в строковый формат
+PrintInet(AF_INET, (const void *)GET_IPV4_DST_ADDR_PTR(p), dst, sizeof(dst)); // #define GET_IPV4_DST_ADDR_PTR(p) ((p)->dst.addr_data32)
+//AF_INET - указывает, что это IPv4 адреса.
+FILE *f = fopen("logs/ip_log.txt", "a");
+
+if (f != NULL &&
+    p->l2.type == PACKET_L2_ETHERNET && // Проверяю, что это Ethernet пакет
+    p->l2.hdrs.ethh != NULL &&  //l2 - уровень канала передачи данных 
+    IsAllowedIP(src))
+{
+    const EthernetHdr *eth = p->l2.hdrs.ethh;
+    
+    fprintf(f, "[IPv4] SRC=%s DST=%s SRC_MAC=%02x:%02x:%02x:%02x:%02x:%02x DST_MAC=%02x:%02x:%02x:%02x:%02x:%02x \n", src, dst,
+        eth->eth_src[0], eth->eth_src[1], eth->eth_src[2],
+        eth->eth_src[3], eth->eth_src[4], eth->eth_src[5],
+
+        eth->eth_dst[0], eth->eth_dst[1], eth->eth_dst[2],
+        eth->eth_dst[3], eth->eth_dst[4], eth->eth_dst[5]);
+    fclose(f);
+}
+/* END CUSTOM */
+
+
     /* save the options len */
     uint8_t ip_opt_len = IPV4_GET_RAW_HLEN(ip4h) - IPV4_HEADER_LEN;
     if (ip_opt_len > 0) {
diff --git a/src/decode-ipv4.h b/src/decode-ipv4.h
index c59eea4cf..deca9d822 100644
--- a/src/decode-ipv4.h
+++ b/src/decode-ipv4.h
@@ -131,6 +131,7 @@ typedef struct IPV4Vars_ {
     uint16_t opt_cnt;
     uint16_t opts_set;
 } IPV4Vars;
+void LoadAllowedIPs(const char *path);
 
 void DecodeIPV4RegisterTests(void);
 
diff --git a/src/suricata.c b/src/suricata.c
index 1d5fcbf05..0b43d4d11 100644
--- a/src/suricata.c
+++ b/src/suricata.c
@@ -3082,6 +3082,15 @@ void SuricataInit(void)
     if (suricata.run_mode == RUNMODE_DPDK)
         prerun_snap = SystemHugepageSnapshotCreate();
 
+/* START CUSTOM */
+    //Выбираю это место в проекте так, как в SuricataInit загружается конфигурация и инициализируются основные компоненты системы.
+    static bool ips_loaded = false; //Позволяет загружать IP-фильтр только один раз
+    if (!ips_loaded) {
+    LoadAllowedIPs("config/ip_filter.conf"); //Загрузка IP-фильтра из файла
+    ips_loaded = true;
+    }
+/* END CUSTOM */
+
     SCSetStartTime(&suricata);
     if (suricata.run_mode != RUNMODE_UNIX_SOCKET) {
         UnixManagerThreadSpawnNonRunmode(suricata.unix_socket_enabled);
